ui/webui/src/components/time/DateAndTime.jsx

```diff
@@ -0,0 +1,476 @@
+                    <FlexItem>
+                        <Button
+                          id={idPrefix + "-edit"}
+                          onClick={() => {
+                              setEditModeEnabled(true);
+                          }}
+                          variant="secondary"
+                        >
+                            {_("Edit")}
+                        </Button>
+                    </FlexItem>
+                </Flex>
+            </CardBody>
+        </Card>
+    );
+};
+
+const RegionSelect = ({ timezones, regions, selectedRegion, setSelectedRegion, setRegionCities }) => {
+    const [isOpenRegionSelect, setIsOpenRegionSelect] = useState(false);
+    const [currentSelection, setCurrentSelection] = useState(null);
+    return (
+        <Select
+          onToggle={(_, isOpen) => setIsOpenRegionSelect(isOpen)}
+          onSelect={(e, selection) => {
+              // set the value from the Select
+              setSelectedRegion(selection);
+              // update the list of cities based on the selected region
+              // TODO: make this dynamic in the CitySelect itself, based on selected region ?
+              setRegionCities(timezones[selection]);
+              // remember it locally so we can show it
+              setCurrentSelection(selection);
+              // then close the Select
+              setIsOpenRegionSelect(false);
+          }}
+          isOpen={isOpenRegionSelect}
+          variant={SelectVariant.typeahead}
+          placeholderText={_("Select region")}
+          selections={currentSelection}
+        >
+            {regions.map(region => <SelectOption
+              key={region.value}
+              value={region} />)}
+        </Select>
+    );
+};
+
+const CitySelect = ({ timezones, regionCities, selectedCity, setSelectedCity }) => {
+    const [isOpenCitySelect, setIsOpenCitySelect] = useState(false);
+    const [currentSelection, setCurrentSelection] = useState(null);
+    return (
+        <Select
+          onToggle={(_, isOpen) => setIsOpenCitySelect(isOpen)}
+          onSelect={(e, selection) => {
+              // set the value from the Select
+              setSelectedCity(selection);
+              // remember it locally so we can show it
+              setCurrentSelection(selection);
+              // then close the Select
+              setIsOpenCitySelect(false);
+          }}
+          isOpen={isOpenCitySelect}
+          variant={SelectVariant.typeahead}
+          placeholderText={_("City")}
+          selections={currentSelection}
+        >
+            {}
+            {regionCities.map(city => <SelectOption
+              key={city.value}
+              value={city} />)}
+        </Select>
+    );
+};
+
+const EditSettingsCard = ({
+    idPrefix,
+    setEditModeEnabled,
+    automaticTimeEnabled,
+    setAutomaticTimeEnabled,
+    timezones,
+    regions,
+    selectedRegion,
+    setSelectedRegion,
+    regionSelected,
```

Comment:
Perhaps not wired in RegionSelect yet (or should selectedRegion be used instead below in RegionSelect?)